src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexPrefix.cs

```diff
@@ -6,16 +6,16 @@ namespace System.Text.RegularExpressions
 {
     internal readonly struct RegexPrefix
     {
-        internal static RegexPrefix Empty { get; } = new RegexPrefix(string.Empty, false);
+        internal static RegexPrefix Empty => new RegexPrefix(string.Empty, caseInsensitive: false);
 
-        internal RegexPrefix(string prefix, bool ci)
+        internal RegexPrefix(string prefix, bool caseInsensitive)
         {
-            Prefix = prefix;
-            CaseInsensitive = ci;
+            Value = prefix;
+            CaseInsensitive = caseInsensitive;
         }
 
-        internal bool CaseInsensitive { get; }
+        internal string Value { get; }
```

Comment:
IMO this should match the constructor parameter -- prefix or value.

Given this is in "character class format" (albeit encoded in weakly typed string) perhaps it woudl be useful to consistently use the same term eg `CharacterClass` ? Other terms are used eg `ToStringClass` on RegexCharClass itself, `GetFirstChars()` seems to always be one of these as well.

Reply:
> Given this is in "character class format"

It's not necessarily, e.g. when looking for a leading substring, that's handed back in this Value.

I'll change the ctor argument name.

Reply:
> It's not necessarily, e.g. when looking for a leading substring, that's handed back in this Value.

Ah, from ComputeLeadingSubstring(). It's confusing to use the same property to expose different "typed" things in different cases. Perhaps that method could return a different struct whose value was a true string, rather than an encoded character class?